File: 2026-03-March-GFG-POTD/March 08 - Pythagorean Triplet/Solution.cpp
class Solution {
public:
    bool pythagoreanTriplet(vector<int>& arr) {
        int n = arr.size();
        
        vector<bool> present(1000001, false);

        // store squares
        for(int i = 0; i < n; i++){
            int sq = arr[i] * arr[i];
            present[sq] = true;
        }

        // check pairs
        for(int i = 0; i < n; i++){
            for(int j = i + 1; j < n; j++){
                int sum = arr[i]*arr[i] + arr[j]*arr[j];
                if(sum <= 1000000 && present[sum]){
                    return true;
                }
            }
        }

        return false;
    }
};

File: 2026-03-March-GFG-POTD/March 08 - Pythagorean Triplet/Solution.py
class Solution:
    def pythagoreanTriplet(self, arr):
        n = len(arr)
        
        present = [False] * 1000001

        # store squares
        for x in arr:
            present[x * x] = True

        # check pairs
        for i in range(n):
            for j in range(i + 1, n):
                s = arr[i]*arr[i] + arr[j]*arr[j]
                if s <= 1000000 and present[s]:
                    return True

        return False
